### PR TITLE
Issue 813 - auto identify build tags and make them available for runtime

### LINKED ISSUE #813: [bug report] internal/build/nocgo.go has typo
Should not use build.Tags = append(build.Tags, "noCGO"), should be Tags = append(Tags, "noCGO"), because it is in the package build

### PR DESCRIPTION
**Pull request by gdey**
Automate the build of the `build.Tags` field.

Fixes #813

### CODE REVIEW

File: internal/build/build_tags.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+	IgnoreCommands bool
+	// IgnorePacakges is a list of packages to ignored
+	IgnorePackages []string
+	// IgnoreTags is a list of tag regular expresions that should be used to ignore tags
+	IgnoreTags []string
+}
+
+// getTags will get the list of packages in the source directory and then return all the build tags
+// in those pacakges.  One can ignore command packages (packages that are called main) by setting
+// ignoreCommands to ture, as well as provide a list of packages to ignore. The list of `ignorePackages`
```
Comment (ARolek) [inline internal/build/build_tags.go:96]: spelling: ture -> true

File: internal/build/build_tags.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+	pkg, err := build.Import(pkgPath, sourceDir, build.IgnoreVendor)
+	if err != nil {
+		return "", "", err
+	}
+	return pkg.Name, pkg.Dir, nil
+}
+
+// BuilTemplateSource is the template used to generate the file
+// we are using the old style go1.6 and below build tag for now
+// till we know for sure that go1.6 is no longer supported
```
Comment (ARolek) [inline internal/build/build_tags.go:165]: till -> untill

File: internal/build/build_tags.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+	}
+	for _, file := range files {
+		if *verbose {
+			fmt.Printf("checking to see if we can remove %v ", file.Name())
+		}
+		if strings.HasSuffix(file.Name(), ".generated.go") {
+			if *verbose {
+				fmt.Printf("yes\n")
+			}
+			os.Remove(filepath.Join(packageDir, file.Name()))
```
Comment (ARolek) [inline internal/build/build_tags.go:250]: capture the error and log it, or explicitly remove ignore it.

**Comment by ARolek** [comment]
This PR looks good overall (a few minor nits in my review). I think it's lacking a reference in a README / HACKING guide though.

**Review by ARolek — APPROVED** [review]
Some minor nits inline, but otherwise looks good. I do think we need to reference this process change in the README / HACKING guide

**Comment by gdey** [comment]
> This PR looks good overall (a few minor nits in my review). I think it's lacking a reference in a README / HACKING guide though.

I added a line to run (go generate ./...) to the Contribruating doc, but there isn't much else that needs to be done. That the main point of this is to simplify the developer experience.

**Review by ARolek — APPROVED** [review]
LGTM!

**Pull request merged**